### PR TITLE
build: Add unifiedpush_ui package

### PR DESCRIPTION
**Pull request by baltevl**
Add unifiedpush_ui as a part of unifiedpush was split off into it.

*Thank you so much for your contribution to FluffyChat ❤️❤️❤️*

Please make sure that your Pull Request meet the following **acceptance criteria**:

- [x] Code formatting and import sorting has been done with `dart format lib/ test/` and `dart run import_sorter:main --no-comments`
- [x] The commit message uses the format of [Conventional Commits](https://www.conventionalcommits.org)
- [x] The commit message describes what has been changed, why it has been changed and how it has been changed
- [ ] (Not applicable) Every new feature or change of the design/GUI is linked to an approved design proposal in an issue
- [ ] (Not applicable) Every new feature in the app or the build system has a strategy how this will be tested and maintained from now on for every release, e.g. a volunteer who takes over maintainership

### Pull Request has been tested on:

- [x] Android (x86, Emulator)
- [ ] iOS
- [ ] Browser (Chromium based)
- [ ] Browser (Firefox based)
- [ ] Browser (WebKit based)
- [ ] Desktop Linux
- [ ] Desktop Windows
- [ ] Desktop macOS

### CODE REVIEW

File: lib/utils/background_push.dart [hunk 1]
```diff
@@ -403,3 +403,26 @@ class BackgroundPush {
     );
   }
 }
+
+class UPFunctions extends UnifiedPushFunctions {
```
Comment (krille-chan) [inline lib/utils/background_push.dart:407]: Hm, this looks somehow generic. Isn't there a default like this in the unifiedpush_ui package itself we could use instead?

Comment (baltevl) [reply lib/utils/background_push.dart:407]: I agree with you but that's the way [up docs](https://unifiedpush.org/developers/flutter/#register-for-pusl) show it. According to [API docs](https://pub.dev/documentation/unifiedpush_ui/latest/unifiedpush_ui/UnifiedPushFunctions-class.html) the methods in question are abstract. So it seems like we need our own implementation :/

**Review by krille-chan — APPROVED** [review]
LGTM thank you very much :)

**Comment by krille-chan** [comment]
@baltevl could you sign your commit? Signed commits is configured as a requirement in this repository

**Comment by baltevl** [comment]
Done :)

**Pull request merged**